Automattic/jetpack
# Tiled Gallery: Ensure container enclosing image is selectable

Fixes https://github.com/Automattic/jetpack/issues/40747

## Issue and Proposed changes:
* Due to a change added to the Gutenberg plugin (in v19.9), the icons to move and remove an image within a Tiled Gallery block were not visible, if the Gutenberg plugin is also active on a site (at v19.9 or greater).
* However, the Tiled Gallery block itself is not built in a way that makes the best sense in terms of interactivity.
* In this PR, I've implemented a temporary fix: moving the interactive focus from the `img` elements to the enclosing `figure` elements, and changing those from `figure` to `div`'s. This prevents the issue mentioned above, though adding the interactive elements to the enclosing `div` is also not a perfect solution. I've added the role of button so that it makes more sense from an accessibility POV.
* As an unrelated small addition, an `aria-hidden="true"` property has been added to the anchor that encloses the Tiled Gallery  `img` in the editor, as the link isn't meant to be functional there.

## For Follow-up:
* I suggest we revisit the Tiled Gallery block and update the process to move and remove images within the block completely, that also considers using drag and drop for re-ordering. I can create another issue to follow up on that as that may require more discussion.
* There may also be possible follow ups on the relevant Gutenberg PR as well - https://github.com/WordPress/gutenberg/pull/67305 - that could continue to affect the block in some way, so I'll continue to follow that.


### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/40747

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:


To test the original behavior:

* On Self-hosted or WoA - ensure the Gutenberg plugin is not installed or active
* Add a Tiled Gallery block to a post or page, and add multiple images.
* On saving the post / page, notice that you can select each image, and move them using the movement arrows.
* You can also delete them using the close icon, and by using the backspace key when the image is selected.

To replicate the issue:

* On Self-hosted and WoA - ensure the Gutenberg plugin is installed and active (v 19.9 or higher (eg. nightly builds)). On Simple, Gutenberg is active by default.
* Revisit the same post / page with the Tiled Gallery. Notice the whole block is now selected, and it isn't possible to select individual images.

To test the fix:
* On self-hosted and WoA - apply this PR using the Jetpack beta plugin (or checkout the PR if testing locally). On Simple, sandbox a Simple site and use the command in the generated comment below.
* Revisit the same post / page with the Tiled Gallery. You should now be able to select and manipulate each image as before.

Existing issues:
* If testing on Simple, there is an existing issue when adding a Tiled Gallery block - after clicking save and refreshing the page you'll see block validation errors in the console (the block itself isn't affected). This occurs on trunk as well as with the branch, but noting it here as it is pre-existing. I only noticed this on Simple.
